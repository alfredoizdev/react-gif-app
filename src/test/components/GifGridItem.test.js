import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Test on <GifGridItem />", () => {
  const title = "some title";
  const url = "http://localhost/some.png";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should render the component ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have p with title coming from props", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("should have img with url as src and title as alt", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("the main div should have class card", () => {
    const div = wrapper.find("div");
    expect(div.prop("className").includes("card")).toBe(true);
  });
});
