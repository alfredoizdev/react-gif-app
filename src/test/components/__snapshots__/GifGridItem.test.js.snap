// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem /> should render the component  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="some title"
    src="http://localhost/some.png"
  />
  <p>
    some title
  </p>
</div>
`;
